### ISSUE TITLE
Failures in persistent >= 2.13 with SafeToRemove columns

### ISSUE THREAD

**Issue by brandonchinn178** [yesodweb/persistent#1273]
My library's tests starting failing when tested against persistent-2.13 using a persistent model with a column marked with `SafeToRemove`.

* In `persistent-sqlite-2.13`, this only seems to break `putMany`.
* In `persistent-postgresql-2.13`, this seems to break `insert` (and since I can't insert, I don't know whether any of the query functions like `get` work)
* In `persistent-postgresql-2.13`, this also seems to break migrations. Previously, if the `SafeToRemove` column exists, running migrations should remove it, but now it seems to try to re-create the column.

CI failure: https://app.circleci.com/pipelines/github/brandonchinn178/persistent-mtl/125/workflows/eb5cdac0-040b-473c-ae93-23a68f88cf1e/jobs/1116

# Repro

This repros the sqlite failure, which is easier to quickly repro than postgres (considering a postgres repro would require running a postgres db). Put the following into a file `Repro.hs` and run `stack Repro.hs`.

```hs
#!/usr/bin/env stack
{-
  stack script
    --resolver nightly-2021-05-20
    --package mtl
    --package monad-logger
    --package persistent
    --package persistent-sqlite
    --extra-dep persistent-2.13.0.1
    --extra-dep persistent-sqlite-2.13.0.0
-}

{-# LANGUAGE DataKinds #-}
{-# LANGUAGE DerivingStrategies #-}
{-# LANGUAGE FlexibleInstances #-}
{-# LANGUAGE GADTs #-}
{-# LANGUAGE GeneralizedNewtypeDeriving #-}
{-# LANGUAGE MultiParamTypeClasses #-}
{-# LANGUAGE OverloadedStrings #-}
{-# LANGUAGE QuasiQuotes #-}
{-# LANGUAGE StandaloneDeriving #-}
{-# LANGUAGE TemplateHaskell #-}
{-# LANGUAGE TypeFamilies #-}
{-# LANGUAGE UndecidableInstances #-}

import Control.Monad.Logger
import Control.Monad.Reader
import Database.Persist
import Database.Persist.Sql
import Database.Persist.Sqlite
import Database.Persist.TH

share [mkPersist sqlSettings, mkMigrate "migration"]
  [persistLowerCase|
Person
  name          String
  age           Int
  removedColumn String SafeToRemove
  UniqueName name
  deriving Show Eq
|]

main :: IO ()
main =
  runStdoutLoggingT $
    withSqliteConn "/tmp/db.sqlite" $ \conn ->
      (`runReaderT` conn) $ do
        _ <- runMigrationSilent migration
        putMany [Person "Alice" 30]
```

Output (with whitespace added by me)
```
[Debug#SQL] CREATE TABLE "person"("id" INTEGER PRIMARY KEY,"name" VARCHAR NOT NULL,"age" INTEGER NOT NULL,CONSTRAINT "unique_name" UNIQUE ("name")); []

[Debug#SQL] INSERT INTO "person"("name", "age", "removed_column") VALUES (?, ?, ?) ON CONFLICT ("name") DO UPDATE SET "name"=EXCLUDED."name", "age"=EXCLUDED."age", "removed_column"=EXCLUDED."removed_column"; [PersistText "Alice",PersistInt64 30]

Foo.hs: SQLite3 returned ErrorError while attempting to perform prepare "INSERT INTO \"person\"(\"name\", \"age\", \"removed_column\") VALUES (?, ?, ?) ON CONFLICT (\"name\") DO UPDATE SET \"name\"=EXCLUDED.\"name\", \"age\"=EXCLUDED.\"age\", \"removed_column\"=EXCLUDED.\"removed_column\"": table person has no column named removed_column
```

**Comment by parsonsmatt** [comment]
good catch, thanks! I've got #1275 up to fix this. Want to try it out and make sure it works well for you?

**Comment by brandonchinn178** [comment]
@parsonsmatt hm it doesn't seem like it works still. Can you try the repro I posted in the original post, changing `persistent-sqlite-2.13.0.0` to `persistent-sqlite-2.13.0.2`?

**Comment by parsonsmatt** [comment]
This is really weird - I can verify that the problem exists with the `stack` script you posted with the new dependency. But I can't figure out why the [test I wrote](https://github.com/yesodweb/persistent/blob/master/persistent-test/src/PersistentTest.hs#L77-L91) with the [very similar model](https://github.com/yesodweb/persistent/blob/master/persistent-test/src/PersistentTestModels.hs#L101-L106) definition passes.

**Comment by parsonsmatt** [comment]
oh lmao i somehow did not actually release the fix. i need to be more careful about my git statusline when doing `stack upload` -_-

**Comment by parsonsmatt** [comment]
OK, trying that above script with `persistent-sqlite-2.13.0.3` and it works.

**Comment by brandonchinn178** [comment]
Do you need to reupload `persistent-postgresql` too?

**Comment by parsonsmatt** [comment]
Is it still broken?

**Comment by brandonchinn178** [comment]
I think so: https://app.circleci.com/pipelines/github/brandonchinn178/persistent-mtl/127/workflows/1cef9026-77a3-4b95-a088-953bc6994240/jobs/1129

**Comment by brandonchinn178** [comment]
@parsonsmatt bump

**Comment by parsonsmatt** [comment]
I released `2.13.0.2` which *should* have the fix, unless I just totally borked it again.

**Comment by brandonchinn178** [comment]
ah yes, that works. thanks!
